### PR TITLE
v11.1.0

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [ ] I've tested my changes, adding unit tests if applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [ ] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

File: autorest/adal/config.go [hunk 1]
```diff
@@ -79,3 +79,50 @@ func NewOAuthConfig(activeDirectoryEndpoint, tenantID string) (*OAuthConfig, err
...
+func NewOAuthConfigWithAPIVersion(activeDirectoryEndpoint, tenantID string, apiVersion *string) (*OAuthConfig, error) {
+	if err := validateStringParam(activeDirectoryEndpoint, "activeDirectoryEndpoint"); err != nil {
+		return nil, err
+	}
+	// it's legal for tenantID to be empty so don't validate it
+	if apiVersion != nil {
+		if err := validateStringParam(*apiVersion, "apiVersion"); err != nil {
+			return nil, err
+		}
+		*apiVersion = fmt.Sprintf("?api-version=%s", *apiVersion)
```
Comment (vladbarosan) [inline autorest/adal/config.go]: won't this mutate the string value that was passed to the function ?

Comment (jhendrixMSFT) [reply autorest/adal/config.go]: D'oh you're correct, will fix.

File: autorest/adal/config.go [hunk 2]
```diff
@@ -79,3 +79,50 @@ func NewOAuthConfig(activeDirectoryEndpoint, tenantID string) (*OAuthConfig, err
 		DeviceCodeEndpoint: *deviceCodeURL,
 	}, nil
 }
+
+// NewOAuthConfigWithAPIVersion returns an OAuthConfig with tenant specific urls.
+// If apiVersion is not nil the "api-version" query parameter will be appended to the endpoint URLs with the specified value.
```
Comment (vladbarosan) [inline autorest/adal/config.go]: can we make the old function call this one to avoid duplicated code ?

Comment (vladbarosan) [reply autorest/adal/config.go]: I presume we would call it with the 1.0 param

Comment (jhendrixMSFT) [reply autorest/adal/config.go]: Derp yep will fix.

**Pull request merged**